src/components/taskDetails/taskDependency/index.tsx

```diff
@@ -1,36 +1,115 @@
-import React, { ChangeEvent, FC, useState } from 'react';
+import React, { ChangeEvent, FC, useEffect, useState } from 'react';
 import { Textarea } from '@/components/taskDetails';
 import { TaskDependencyProps } from '@/interfaces/taskDetails.type';
 import DependencyList from '@/components/taskDetails/taskDependency/DependencyList';
-import { parseDependencyValue } from '@/utils/parseDependency';
+import useDebounce from '@/hooks/useDebounce';
+import { useGetAllTasksQuery } from '@/app/services/tasksApi';
+import classNames from '../task-details.module.scss';
+
+interface Task {
+    id: string;
+    title: string;
+}
+
+const TaskInput: FC<{
+    searchTerm: string;
+    handleChange: (
+        event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+    ) => void;
+}> = ({ searchTerm, handleChange }) => (
+    <Textarea
+        name="dependsOn"
+        value={searchTerm}
+        onChange={handleChange}
+        testId="dependency-textarea"
+        placeholder="Search for task"
+    />
+);
```

Comment:
NIT -Please make this a sperate component

Reply:
As it is not usable anywhere so it is unnecessary to create separate file for this